### PR TITLE
Update to new ctl

### PR DESCRIPTION
**Pull request by marcusbfs**
Update CTL to v9.1.0, whic provides basic Conway support and replaces Plutip with `cardano-testnet`.

### Prereview checklist

- [ ] The [CHANGELOG.md](../blob/master/CHANGELOG.md) has been updated under the `# Unreleased` header using the appropriate sub-headings with links to the appropriate issues/PRs.
- [x] All tests pass in CI
- [ ] PR was self-reviewed

### CODE REVIEW

**Review by jstolarek — COMMENTED** [review]
I've read through the code so far and have some questions - see inline comments.

I note that this update introduces hundreds of warnings in the form of:
```
  A custom warning occurred while solving type class constraints:

    Contract.TxConstraints is deprecated. Use `purescript-cardano-transaction-builder`

  in value declaration mintNonAdaTokens
```
I understand that fixing them is outside of the scope of this PR, but we should silence those warnings ASAP to make sure no really relevant warnings get lost in the noise.

All that being said, I was unable to run the code - neither the test suite nor the executable itself. Every attempt at doing so resulted in an error:

```
[killy@skynet : /dane/projekty/binarapps/trustless-sidechain/offchain] spago test
           Src   Lib   All
Warnings   0     0     0
Errors     0     0     0
[info] Build succeeded.
file:///dane/projekty/binarapps/trustless-sidechain/offchain/output/Cardano.Serialization.Lib/foreign.js:452
  CSL.DrepVotingThresholds.new_default();
                           ^

TypeError: CSL.DrepVotingThresholds.new_default is not a function
    at file:///dane/projekty/binarapps/trustless-sidechain/offchain/output/Cardano.Serialization.Lib/foreign.js:452:28

Node.js v20.12.2
[error] Tests failed: exit code: 1
```
Thus I can't say anything about the transition from Plutip to Testnet nor actual compatibility with the post-HF network.

File: offchain/test/Test/Config.purs [hunk 1]
```diff
@@ -29,7 +27,7 @@ config =
   , clusterConfig:
       { slotLength: Seconds 0.05
       , epochSize: Nothing
-      , maxTxSize: Nothing
-      , raiseExUnitsToMax: false
+      , testnetMagic: 2
+      , era: Babbage
```
[the remark this answers is not in the document]
Comment (klntsky) [reply offchain/test/Test/Config.purs:31]: cardano-testnet does not support Conway yet

File: offchain/test/Test/Config.purs [hunk 2]
```diff
@@ -2,15 +2,13 @@ module Test.Config (config) where
 
 import Contract.Prelude
 
-import Contract.Test.Plutip (PlutipConfig)
+import Ctl.Internal.Testnet.Types (Era(Babbage), TestnetConfig)
```
[the remark this answers is not in the document]
Comment (klntsky) [reply offchain/test/Test/Config.purs:5]: If it's not possible to import from a public module, we should modify CTL @marcusbfs

File: offchain/src/TrustlessSidechain/Effects/Transaction.purs [hunk 3]
```diff
@@ -53,6 +61,61 @@ import TrustlessSidechain.Error (OffchainError)
...
 import Type.Row (type (+))
 
+-- | A newtype for the unbalanced transaction after creating one with datums
+-- | and redeemers not attached.
+newtype UnbalancedTx = UnbalancedTx
+  { transaction ∷ Transaction -- the unbalanced tx created
+  , usedUtxos ∷ Map TransactionInput TransactionOutput
+  , datums ∷
+      Array PlutusData -- the array of ordered datums that require attaching
+  , redeemers ∷ Array UnindexedRedeemer
```
Comment (klntsky) [inline offchain/src/TrustlessSidechain/Effects/Transaction.purs:71]: These are unused, just `transaction` and `usedUtxos` are enough, if you want to preserve the old behavior.

File: offchain/src/TrustlessSidechain/Error.purs [hunk 4]
```diff
@@ -7,7 +7,7 @@ import Contract.Prelude
 import Cardano.Types.ScriptHash (ScriptHash)
 import Contract.Address (Address)
 import Contract.Transaction as Transaction
-import Contract.UnbalancedTx (MkUnbalancedTxError)
+import Ctl.Internal.ProcessConstraints.Error (MkUnbalancedTxError)
```
[the remark this answers is not in the document]
Comment (marcusbfs) [reply offchain/src/TrustlessSidechain/Error.purs:10]: In the latest CTL version, `MkUnbalancedTxError` is now internal. However, using an internal module in this context is not a concern.

File: offchain/src/TrustlessSidechain/Effects/Transaction.purs
[the remark this answers is not in the document]
Comment (marcusbfs) [reply offchain/src/TrustlessSidechain/Effects/Transaction.purs:1]: Yes, the code relies on a removed data structure. However, as Vladimir mentioned, there are unused fields that we can remove.

**Pull request closed without merging**